### PR TITLE
use a separate func instead of replacing the diagnostics messages

### PR DESCRIPTION
**Pull request by abzcoding**
# Description

Changing diagnostics messages would cause some weird issues regarding code actions

here is the result
image

### CODE REVIEW

**Review by ChristianChiarulli — APPROVED** [review]

**Comment by ChristianChiarulli** [comment]
tested it, seems fine to me

File: lua/lsp/handlers.lua [hunk 1]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+      width = string.len(diags[i].message)
+    end
+  end
+
+  opts = vim.lsp.util.make_floating_popup_options(width, height, opts)
+  opts["style"] = "minimal"
+  opts["border"] = "rounded"
+
+  vim.api.nvim_buf_set_option(bufnr, "bufhidden", "wipe")
+  local winnr = vim.api.nvim_open_win(bufnr, false, opts)
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:85]: When hitting gl on lines without diagnostics

image

File: lua/lsp/handlers.lua [hunk 2]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+  local close_events = { "CursorMoved", "CursorMovedI", "BufHidden", "InsertCharPre" }
+  local bufnr = vim.api.nvim_create_buf(false, true)
+  local diagnostic_severities = {
+    "Error",
+    "Warning",
+    "Information",
+    "Hint",
+  }
+
+  for i, v in ipairs(diagnostics) do
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:61]: Please rename `v` to `diagnostic`

File: lua/lsp/handlers.lua [hunk 3]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+    if source then
+      if string.find(source, "/") then
+        source = string.sub(v.source, string.find(v.source, "([%w-_]+)$"))
+      end
+      diags[i].message = string.format("%s: %s", source, v.message)
+    else
+      diags[i].message = string.format("%s", v.message)
+    end
+
+    if vim.tbl_contains(vim.tbl_keys(v), "code") then
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:72]: `if diagnostic.code then`

File: lua/lsp/handlers.lua [hunk 4]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+      diags[i].message = diags[i].message .. string.format(" [%s]", v.code)
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:73]: A detail, but I would write it `string.format("%s [%s]", diags[i].message, v.code)`
I like to see the complete format and not a mix of concatenated strings :)

File: lua/lsp/handlers.lua [hunk 5]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+    end
+    if string.len(diags[i].message) > width then
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:75]: `diags[i].message:len()`

File: lua/lsp/handlers.lua [hunk 6]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+
+  opts = vim.lsp.util.make_floating_popup_options(width, height, opts)
+  opts["style"] = "minimal"
+  opts["border"] = "rounded"
+
+  vim.api.nvim_buf_set_option(bufnr, "bufhidden", "wipe")
+  local winnr = vim.api.nvim_open_win(bufnr, false, opts)
+  vim.api.nvim_win_set_option(winnr, "winblend", 0)
+  vim.api.nvim_buf_set_var(bufnr, "lsp_floating_window", winnr)
+  for i, v in ipairs(diags) do
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:88]: v -> diag

File: lua/lsp/handlers.lua [hunk 7]
```diff
@@ -60,4 +43,64 @@ function M.setup()
...
+    vim.api.nvim_buf_set_lines(bufnr, i - 1, i - 1, 0, { v.message })
+    vim.api.nvim_buf_add_highlight(
+      bufnr,
+      -1,
+      "LspDiagnosticsFloating" .. diagnostic_severities[v.severity],
+      i - 1,
+      0,
+      string.len(v.message)
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:96]: :len

File: lua/lsp/handlers.lua [hunk 8]
```diff
@@ -60,4 +43,64 @@ function M.setup()
   })
 end
 
+function M.show_line_diagnostics()
+  local diagnostics = vim.lsp.diagnostic.get_line_diagnostics()
+  local diags = vim.deepcopy(diagnostics)
+  local height = #diagnostics
```
Comment (Tastyep) [inline lua/lsp/handlers.lua:49]: Maybe return if height is 0

**Review by Tastyep — CHANGES_REQUESTED** [review]

**Comment by abzcoding** [comment]
@Tastyep thank you for the review, can you test again?

**Review by Tastyep — APPROVED** [review]
I also tested with multiple diagnostics on the same line, it works well :+1:

**Review by kylo252 — APPROVED** [review]
This works as expected 😍 

We can always iterate later on the underlying function, i.e. remove the hard-coded border style and close events.

**Pull request merged**